File: 392-is-subsequence/is-subsequence.c

bool isSubSequence(char *strA, char *strB, int m, int n)
{
	//strA - Big sting
	//strA - Small sting

	if(n==0) 
		return true;
	if(m==0) // nothing to compare and  n>0 ...so false
		return false;

	if(strA[m-1]==strB[n-1])
		return isSubSequence(strA, strB, m-1, n-1);
	else
		return isSubSequence(strA, strB, m-1, n);//only decrement m to see StrB is present in StrA
}
bool isSubsequence(char* s, char* t) {
    return isSubSequence(t,s, strlen(t), strlen(s));
}

File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequenceHelper(self, s, t, m, n) -> bool:

        if n==0 :
            return True
        if m==0 :
            return False
        
        if s[m-1] == t[n-1]:
            return self.isSubsequenceHelper(s, t, m-1, n-1)
        else:
            return self.isSubsequenceHelper(s, t, m-1, n)

    def isSubsequence(self, s, t):
        return self.isSubsequenceHelper(t, s, len(t), len(s))